src/main/java/com/ifmo/epampractice/daoimpl/CarModelDaoImpl.java

```diff
@@ -0,0 +1,115 @@
+package com.ifmo.epampractice.daoimpl;
+
+import com.ifmo.epampractice.dao.CarModelDao;
+import com.ifmo.epampractice.dao.DBConnectorPostgres;
+import com.ifmo.epampractice.entity.CarModelEntity;
+
+import java.sql.*;
+import java.util.LinkedList;
+import java.util.List;
+
+public class CarModelDaoImpl implements CarModelDao {
+    private DBConnectorPostgres dbConnector = DBConnectorPostgres.getInstance();
+
+    @Override
+    public List<CarModelEntity> getAll() {
+        final String sql = "SELECT * FROM car_model";
+        List<CarModelEntity> carModelEntityList = new LinkedList<>();
+        try (
+                Connection connection = dbConnector.getConnection();
+                Statement statement = connection.createStatement()
+        ) {
+            ResultSet rs = statement.executeQuery(sql);
+            while (rs.next()) {
+                carModelEntityList.add(entityFromResultSet(rs));
+            }
+        }
+        catch (SQLException e) {
+            e.printStackTrace();
+        }
+        return carModelEntityList;
+    }
+
+    @Override
+    public CarModelEntity get(int id) {
+        final String sql = "SELECT * FROM car_model WHERE id = ?";
+        try (
+                Connection connection = dbConnector.getConnection();
+                PreparedStatement statement = connection.prepareStatement(sql)
+        ) {
+            statement.setInt(1, id);
+            ResultSet rs = statement.executeQuery();
+            if (!rs.next()){
+                return null;
+            }
+            return entityFromResultSet(rs);
+        }
+        catch (SQLException e) {
+            e.printStackTrace();
+            return null;
+        }
+    }
+
+    @Override
+    public void update(CarModelEntity o) {
+        final String sql = "UPDATE car_model SET brand_name = ?, model_name = ?, " +
+                "price_per_hour = ? WHERE id = ?";
+        try (
+                Connection connection = dbConnector.getConnection();
+                PreparedStatement statement = connection.prepareStatement(sql)
```

Comment:
No need to close statement because it's automatically closed when parent Connection is closed.

Reply:
Actually it's not (my bad, sorry).
ResultSet is indeed closed when parent Statement is closed (https://docs.oracle.com/javase/8/docs/api/java/sql/ResultSet.html).
Statement, on the other hand, is NOT guaranteed to be closed after it's parent Connection is closed since Java 7.

I guess it's best to close all of them (Connection, Statement, ResultSet) just to be safe.

Reply:
@Aeeee done. But I think that we need to close only Connection and Statement, because sometimes close ResultSet after setting Statement fields is difficult and can breaks codestyle